uamqp/async_ops/connection_async.py

```diff
@@ -98,6 +99,15 @@ def __init__(self, hostname, sasl,
         await self.destroy_async()
         _logger.debug("Finished exiting connection %r context.", self.container_id)
 
+    async def _open_async(self):
+        self._conn.open()
+        connection_state = c_uamqp.ConnectionState(self._conn.get_state())
+        while connection_state != c_uamqp.ConnectionState.OPENED:
```

Comment:
I feel like this could be made more resiliant... perhaps something like:
```
while connection_state not in [OPENED, ERROR, etc etc]
# or the reverse
while connection_state in [list of all the opening states]
```

Reply:
Same comment for sync connection

Reply:
Code updated.
also added new parameter `open_timeout` in the connection constructor and `timeout` in `open` function

as @KieranBrantnerMagee mentioned his concern on the looping forever in the `open` function. I checked the underlying `socketio_dowork` but find nothing related to timeout control.